### PR TITLE
fix azure_rm_subnet_info when listing all subnets

### PR DESCRIPTION
**Pull request by ronansalmon**
##### SUMMARY
- fix the following error :
 An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: get() takes at least 4 arguments (3 given)

- fix ansible_facts variable

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
azure_rm_subnet_facts.py

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_subnet_facts.py [hunk 1]
```diff
@@ -53,11 +53,10 @@
     azure_rm_subnet_facts:
       resource_group: myResourceGroup
       virtual_network_name: myVirtualNetwork
-      name: mySubnet
 '''
 
 RETURN = '''
-subnets:
+azure_subnets:
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/azure/azure_rm_subnet_facts.py]: Same here: never change return values.

File: lib/ansible/modules/cloud/azure/azure_rm_subnet_facts.py [hunk 2]
```diff
@@ -149,21 +148,14 @@ class AzureRMSubnetFacts(AzureRMModuleBase):
...
-            )
+            name=dict(type='str'),
+            resource_group=dict(required=True, type='str'),
+            virtual_network_name=dict(required=True, type='str'),
         )
         # store the results of the module operation
         self.results = dict(
-            changed=False
+            changed=False,
+            ansible_facts=dict(azure_subnets=[])
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/azure/azure_rm_subnet_facts.py]: Also, don't return `ansible_facts` from cloud modules.

**Review by felixfontein — CHANGES_REQUESTED** [review]

**Comment by Fred-sun** [comment]
@ronansalmon Thank for your contribution, Please help resolve the shippable fail when you're free! Thank you very much!

**Comment by ronansalmon** [comment]
Will send a patch when i'll get back in 3 weeks.

File: lib/ansible/modules/cloud/azure/azure_rm_subnet_facts.py (hunk 1 above)
Comment (ronansalmon) [reply lib/ansible/modules/cloud/azure/azure_rm_subnet_facts.py]: fix in last commit

**Comment by Fred-sun** [comment]
@yungezz @MyronFanQiu Please take a look this PR when you are free! ready_for_review, Thank you very much!

**Comment by Fred-sun** [comment]
@zikalino @yungezz ready_for_review

**Comment by ronansalmon** [comment]
I have updated the path to be compatible with azure_rm_subnet_info

**Comment by felixfontein** [comment]
@ronansalmon please ask someone from the Azure team, I'm not familiar with Azure at all.

BTW, all bugfixes need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to). So you should definitely add one.

**Review by felixfontein — COMMENTED** [review]
Needs changelog fragment. Can't comment besides that.

**Comment by ronansalmon** [comment]
Ok, I've pushed a changelog.
@felixfontein, as far as I can understand, this PR says that you have requested a change. What needs to be changed ?
Ronan

**Comment by felixfontein** [comment]
@ronansalmon if you scroll up, then you can see that on June 28, I request some changes with regard to return values. Since these changes have been addressed, I'm happy.

**Review by felixfontein — APPROVED** [review]
Ok with regard that no `ansible_facts` are returned. This was a problem with an earlier version of the PR. I cannot comment on anything else; this should be reviewed by the Azure team.

**Comment by felixfontein** [comment]
(It seems to be hard to get rid of the "request changes"...)

**Comment by Fred-sun** [comment]
@yungezz @zikalino The PR change small, please review and push for merge! Thank you very much!

**Comment by Fred-sun** [comment]
ping to merge

**Comment by Fred-sun** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@haiyuazhang merge or not? Please check! Thank you very much!

**Comment by Fred-sun** [comment]
@haiyuazhang @gavinfish ping to merge!

**Comment by Fred-sun** [comment]
@haiyuazhang @gavinfish @zikalino Ping to merge!

**Comment by Fred-sun** [comment]
@ronansalmon Thank you for taking the time to contribute to this PR. We will transfer ansible 's azure module related Issue and PR to azure collection (https://github.com/ansible-collections/azure/pulls), can you transfer the Issue to azure collection repo?

**Comment by ronansalmon** [comment]
Unfortunately I won't bother. I don't think that this issue will ever be merged. The Azure team didn't care about this minor patch for the last 6 month. They didn't event bother replying to your multiple calls:(

So, I gave up using Azure and Ansible's azure module. I'm still using Ansible though, and will still try to contribute !

@Fred-sun, thanks for trying.

**Pull request closed without merging**